syntax = "proto3";

package NAlice;

option go_package = "a.yandex-team.ru/alice/begemot/lib/frame_aggregator/proto";
option java_package = "ru.yandex.alice.alice.begemot.lib.frame_aggregator.proto";

message TFrameAggregatorConfig {
    message TClassifierRule {
        string Source = 1;
        string Model = 2; // used in AliceWizDetection and AliceMultiIntentClassifier sources
        string Intent = 3;
        float Threshold = 4;
        float Confidence = 5;
        bool IsNegative = 6;
        optional float LoggingThreshold = 7;
        optional string Anchor = 8;
    }

    message TTaggerRule {
        string Source = 1;
        string Intent = 2;
    }

    message TRule {
        TClassifierRule Classifier = 1;
        optional TTaggerRule Tagger = 2;
        repeated string Experiments = 3;
    }

    message TFrame {
        string Name = 1;
        repeated TRule Rules = 2;
        repeated string Experiments = 3;
    }

    repeated TFrame Frames = 1;
    string Language = 2;
}
